main.js

```diff
@@ -95,6 +95,53 @@
       h: parseInt(array[3].trim(), 10)
     };
   }
+
+  function toggleDummyCanvas() {
+    var subCanvas = document.getElementById('sub-canvas');
```

Comment:
sorry, what does shadowing mean by?
- By creating another layer that drawing sub-lines
- Switching hide and show with sub-lines layer